k8s/gitlab/README.md

```diff
@@ -353,6 +353,117 @@ echo "http://${SERVICE_IP}/"
 
 Use your GitLab root password generated earlier to sign-in.
 
+# Backup
+
+Below steps are based on [Official GitLab Backup/Restore documentation](https://docs.gitlab.com/ee/raketasks/backup_restore.html#back-up-gitlab) .
+
+```
+APP_INSTANCE_NAME=gitlab-1
+NAMESPACE=default
+```
+
+Run Backup command:
+```
+kubectl --namespace ${NAMESPACE} exec -it \
+  "${APP_INSTANCE_NAME}-gitlab-0" -- bash -c "gitlab-backup create"
+```
+
+You can see created backup files from output of backup command or via running:
+```
+kubectl --namespace ${NAMESPACE} exec -it \
+  "${APP_INSTANCE_NAME}-gitlab-0" -- bash -c \
+  "ls /var/opt/gitlab/backups/"
+```
+
+Copy backup file to local workstation:
+```
+BACKUP_TAR="place-your-backup-file-here"
+
+kubectl --namespace ${NAMESPACE} cp \
+  ${APP_INSTANCE_NAME}-gitlab-0:/var/opt/gitlab/backups/${BACKUP_TAR} .
+```
+
+> GitLab does not back up any configuration files, SSL certificates, or
+> system files.  You are highly advised to read about storing
+> configuration files:
+> https://docs.gitlab.com/ee/raketasks/backup_restore.html#storing-configuration-files
+
+Copy secret files manually:
+```
+for secret in "gitlab-secrets.json" "gitlab.rb"; do
+  kubectl --namespace ${NAMESPACE} cp ${APP_INSTANCE_NAME}-gitlab-0:/etc/gitlab/$secret .
+done
+```
+
+Save your root password of this database to local workstation.
+
+```
+kubectl --namespace ${NAMESPACE} get secret \
+  ${APP_INSTANCE_NAME}-gitlab-secret \
+  -ojsonpath='{.data.gitlab-root-password}' > root-password-base64.txt
+```
+
+# Restore
+For Gitlab Restore Prerequisites visit [here](https://docs.gitlab.com/ee/raketasks/backup_restore.html#restore-prerequisites).
+
+Below steps are based on [Official GitLab Backup/Restore documentation](https://docs.gitlab.com/ee/raketasks/backup_restore.html#restore-for-omnibus-gitlab-installations).
+
+We assume you have below files in local workstation if you followed backup procedure in this document.
+
+```
+<STAMP>_gitlab_backup.tar 	# Gitlab Backup tar file
+
+gitlab.rb 					# Database encryption key related secret files.
+gitlab-secrets.json         # For more information visit https://docs.gitlab.com/ee/raketasks/backup_restore.html#storing-configuration-files
+
+root-password-base64.txt   	#File contains encoded root password.
+```
```

Comment:
Looks weird because of inconsistent spaces/tabs.
As this is not a code snipped, imho better to use regular bullet list as below:

- `<STAMP>_gitlab_backup.tar` - Gitlab Backup tar file
- `gitlab.rb` - Database encryption key related secret files.
- `gitlab-secrets.json` - For more information visit https://docs.gitlab.com/ee/raketasks/backup_restore.html#storing-configuration-files
- `root-password-base64.txt` - File contains encoded root password.